### PR TITLE
Person form accepts phones and emails

### PR DESCRIPTION
**Pull request by corincerami**
This sets both `phones` and `emails` as a `has_many` relationship on `Person`. This allows `Person` to accept nested attributes in the new person form. Phone and Email are now STI types of Channels, since they both already had their own subclasses.

This PR also adds strong params.

### CODE REVIEW

**Comment by corincerami** [comment]
@kgf This PR is ready for review.

File: Gemfile [hunk 1]
```diff
@@ -29,6 +30,7 @@ group :test, :development do
   gem 'database_cleaner', '1.0.1'
   gem 'test-unit'
   gem 'poltergeist'
+  gem 'pry'
```
Comment (kgf) [inline Gemfile]: Not for this PR, but pry-nav might be a good (better? ) addition.

Comment (marnen) [inline Gemfile]: Or byebug.

File: app/models/person.rb [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
+  include ActiveModel::ForbiddenAttributesProtection
   before_save :title_order
 
-  attr_accessible :firstname, :lastname, :status, :icsid, :department, :city, :state, :zipcode, :start_date, :end_date , :title, :gender, :date_of_birth,:division1, :division2, :channels_attributes, :title_ids, :title_order, :comments
-
-  #Having a condition on this association allows all the chaining magic to happen.
-  #Could I use a named scope, and/or could I have another association for 'active_certs' ?
   has_many :certs, :conditions => {:status =>'Active' }
-
-  # Since phones and emails derive off of channels, is "has_many :channels" below redundant?
```
Comment (kgf) [inline app/models/person.rb:12]: Chris, you deleted the comment/question, and left the `has_many` call. Do you know the answer ?

File: app/models/person.rb [hunk 3]
```diff
@@ -23,7 +22,6 @@ class Person < ActiveRecord::Base
 
   validates_presence_of :firstname, :lastname, :status
   validates_uniqueness_of :icsid, :allow_nil => true, :allow_blank => true   # this needs to be scoped to active members, or more sophisticated rules
-  validates :firstname, :uniqueness => { :scope => :lastname }
```
Comment (kgf) [inline app/models/person.rb:26]: Why was this removed ?

File: app/controllers/people_controller.rb [hunk 4]
```diff
@@ -88,6 +88,8 @@ def new
     @page_title = "New Person"
     @person = Person.new(status: cookies[:status], state: 'MA')
     @person.channels.build (attributes = {category: 'E-Mail', status: 'OK', usage: '1-All'})
+    @person.phones.build
+    @person.emails.build
     @mobile = @person.channels.build (attributes = {category: 'Mobile Phone', status: 'OK', usage: '1-All'})
```
Comment (kgf) [inline app/controllers/people_controller.rb]: I would think we don't need both this and line 92. These lines are to build objects to be populated from the form. Also, these should only be on the new action, not the edit.

File: app/controllers/people_controller.rb [hunk 5]
```diff
@@ -102,8 +104,8 @@ def edit
   end
```
Comment (kgf) [inline app/controllers/people_controller.rb:103]: On line 103, we should make sure that phones and emails are handed into the form. I believe we are broken right now.

File: app/models/person.rb (hunk 3 above)
Comment (corincerami) [inline app/models/person.rb:26]: What if you have 2 `John Smith`s?

File: app/models/person.rb (hunk 2 above)
Comment (corincerami) [inline app/models/person.rb:12]: has_many channels is necessary if you ever want to check all of a `person`'s channels, without thinking about whether they are `phone`s or `email`s or whatnot. `person.channels` will give you all of the means of contacting a person. I'll let you decide if you think that's valuable or not.

Comment (kgf) [inline app/models/person.rb:12]: Thanks. I intend to keep it.

File: db/schema.rb [hunk 6]
```diff
@@ -189,23 +188,6 @@
     t.datetime "updated_at",     :null => false
   end
 
-  create_table "notifications", :force => true do |t|
```
Comment (kgf) [inline db/schema.rb:192]: I'm confused by these changes to schema. It looks like you are removing multiple tables that deal with notifying people ?

Comment (corincerami) [inline db/schema.rb:192]: Hm, I'm not sure where those schema changes came from. I'll fix the PR.

**Comment by corincerami** [comment]
@kgf I've fixed the schema.rb

**Comment by houhoulis** [comment]
I'm sorta guessing the PR doesn't need a change any more  :smiley_cat:

**Comment by houhoulis** [comment]
thanks @chrisccerami !
